### DISCUSSION TITLE
Q: the Global Strain method in simple Applied strain problem instead of applied stress.

### DISCUSSION THREAD

**Discussion by Ttw0626** [idaholab/moose#24495] [Q&A Modules: Solid mechanics] [answered]
Dear Moose developers,

### **1) Reason:**
I totally understand **the applied stress simple PBC test** [x y z all of periodic property] using **Global Strain**.
There are many tests and instructive examples in " /moose/test/tests/ " and " /TM/test/tests/ " and "/combine/example", Thanks for that.

### **2) Q1:**
However, I wanna ask how to apply the strain-control on global excepting the calculated eigenstrain in coupled field example ?? For example, I wanna apply **the volumetric e00 = 0.001** along X direction and all of directions maintain periodic.

### **3) Q2:**
Or I should use **Mortar periodic constraint** and **Lagrangian new system** to do that ?

kind regards,
TAN

**Comment by hugary1995** [comment, marked as answer]
You can use the new homogenization system that comes with the new Lagrangian kernels (`new_system = true` in the TM action). Examples are inside `modules/tensor_mechanics/test/tests/lagrangian/cartesian/total/homogenization`. It's a more flexible system than the original global strain system.

The key object is the homogenization userobject:
```python
[UserObjects]
  [homogenization]
    type = HomogenizationConstraint
    constraint_types = 'STRAIN NONE NONE STRESS STRESS NONE STRAIN STRESS STRESS'
    targets = '0.1 s s e s s'
    execute_on = 'INITIAL LINEAR NONLINEAR'
  []
[]
```

The parameter `constrain_types` specifies which type of constraint you want to apply for each component. They are ordered column major IIRC. You need to specify 9 components for large deformation and 6 components for small deformation. `STRAIN` means you want to constrain the homogenized strain, `STRESS` means you want to constrain the homogenized stress, and `NONE` means no constraint should be applied for that component.

Any non-`NONE` constraint type should correspond to a target value specified by the `targets` parameter. Each target is a Function. You can also specify a numeric literal. In the above example, `0.1` means you want to constrain the homogenized strain_xx to be 0.1.

Making the homogenization system documentation more complete is on my to-do list.

**Reply by Ttw0626** [reply]
Thanks, gray.

I got it, I will try.

TAN
